### ISSUE TITLE
Issue running tests request.py

### ISSUE THREAD

**Issue by enicky** [alistair23/AutoMower-BLE#17]
When running the tests of the request file, I run into some issues: 
```
======================================================================
FAIL: test_generate_request_mode_of_operation (__main__.TestStringMethods.test_generate_request_mode_of_operation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/enicky/project/AutoMower-BLE/automower_ble/request.py", line 484, in test_generate_request_mode_of_operation
    self.assertEqual(
AssertionError: b'02fd11001aca9857013400afea110100010000fe03' != b'02fd11001aca9857013400afea1100000100003303'

----------------------------------------------------------------------
```

The difference is in the last appended byte before the crc. The 'fe' gets added and you expect '33' ? 

Are you having the same issue ? 

I encountered this issue when trying to get this code running for my Gardena mower. It appears that I am able to retrieve data from the mower, but all commands that are send to start / stop / ... are being accepted (I guess), I also get a response after that command. But the mower isn't reacting to it. So I thought to run those tests ... and then I encountered this issue. 

Is everyone having this issue ? Is the assert checking on an incorrect result ?

**Comment by erikmol** [comment]
Yes I also encounter the same error. Moreover, the "fe"/"33" is the CRC, but there is another difference if you look more closely.

```
02 fd 11 00 1a ca 98 57 01 34 00 af ea 11 01 00 01 00 00 fe 03
02 fd 11 00 1a ca 98 57 01 34 00 af ea 11 00 00 01 00 00 33 03
------------------------------------------^^
```

This bytearray indicates that the command sent is `(4586, 0)` with data `0`. This command is not implemented and thus this test will always fail.

> I encountered this issue when trying to get this code running for my Gardena mower. It appears that I am able to retrieve data from the mower, but all commands that are send to start / stop / ... are being accepted (I guess), I also get a response after that command. But the mower isn't reacting to it. So I thought to run those tests ... and then I encountered this issue.

From what I've seen in other logs, for any request there is always a response. The response might not carry any data though. Not sure if this brings any clarity?

**Comment by alistair23** [comment]
I don't have a CI setup yet, so it's possible the data has changed and the test didn't. 

It looks like it should be `generate_request_mode_of_operation("auto")` in the test case

**Comment by enicky** [comment]
I want to start looking into why the code isn't working for the Minimo. But should I continue with your code @alistair23  ? Or ... is it better to continue with @erikmol 's code ? Cause I saw that it really is a complete refactoring ... ?

**Comment by alistair23** [comment]
Ideally it shouldn't matter which code you use. The goal is that at the end it's the same requests/responses. So check the phone app against either

**Comment by enicky** [comment]
Got the code up and running for my Minimo 400. I'll create   [PR](https://github.com/alistair23/AutoMower-BLE/pull/22) it. Added a lot more commands to simulate every call my phone made to the mower.

What I "think" what did it was adding the `send_operator_pin_request` before doing anything else. (already implemented by @erikmol  )
